foundation-extensions/experience-platform-extension/edge-network-api-reference.md

```diff
@@ -278,7 +350,325 @@ public class EdgeEventHandle {
 }
 ```
 
-Use this class when calling the sendEvent API with EdgeCallback.
+{% endtab %}
+
+{% tab title="iOS (AEP 3.x)" %}
+
+### Swift
+
+```swift
+@objc(AEPEdgeEventHandle)
+public class EdgeEventHandle: NSObject, Codable {
+
+    /// Encodes the event to which this handle is attached as the index in the events array in EdgeRequest
+    internal let eventIndex: Int?
+
+    /// Payload type
+    @objc public let type: String?
+
+    /// Event payload values
+    @objc public let payload: [[String: Any]]?
+}
+```
+{% endtab %}
+{% endtabs %}
+
+Use this class when calling the [sendEvent](edge-network-api-reference.md#sendevent) API with EdgeCallback.
+
+### ExperienceEvent
+
+Experience Event is the event to be sent to Adobe Experience Platform Edge Network.
+The XDM data is required for any Experience Event being sent using the Edge extension.
+
+{% tabs %}
+{% tab title="Android" %}
+
+### Java
+
+```Java
+public final class ExperienceEvent {
+
+  
+public Builder() {...}
+
+  /**
+    * Sets free form data associated with this event to be passed to Adobe Experience Edge.
+    *
+    * @param data free form data, JSON like types are accepted
+    * @return instance of current builder
+    * @throws UnsupportedOperationException if this instance was already built
+    */
+  public Builder setData(final Map<String, Object> data) {...}
+
+  /**
+    * Solution specific XDM event data for this event.
+    * If XDM schema is set multiple times using either this API or {@link #setXdmSchema(Map)},
+    * the value will be overwritten and only the last changes are applied.
+    * Setting {@code xdm} to null clears the value.
+    *
+    * This event is sent to the Experience Platform dataset defined by {@link Schema#getDatasetIdentifier()}.
+    *
+    * @param xdm {@link Schema} information
+    * @return instance of current builder
+    * @throws UnsupportedOperationException if this instance was already built
+    */
+  public Builder setXdmSchema(final Schema xdm) {...}
+
+  /**
+    * Solution specific XDM event data for this event, passed as raw mapping of keys and
+    * Object values.
+    * If XDM schema is set multiple times using either this API or {@link #setXdmSchema(Schema)},
+    * the value will be overwritten and only the last changes are applied.
+    * Setting {@code xdm} to null clears the value.
+    *
+    * When using {@link Edge#sendEvent}
+    * this event is sent to the Experience Platform {@code datasetIdentifier} if provided,
+    * or to the default Experience Platform dataset defined when generating the Experience Edge
+    * configuration ID if {@code datasetIdentifier} is null.
```

Comment:
I feel like we can simplify this class docs so we can focus more on the public APIs, if needed all these details are included with the javadocs. 
I am thinking we can keep max one short line description here and the params/returns description only similar with the Swift one. @cacheung @kevinlind thoughts?

Reply:
I simplify the description, please check if it looks good.